src/RageMath.cpp

```diff
@@ -616,8 +616,8 @@ float RageFastSin(float angle)
 {
 	if(angle == 0) { return 0; }
 	float index= angle * sine_table_index_mult;
-	int first_index= static_cast<int>(index);
-	int second_index= (first_index + 1) % sine_index_mod;
+	auto first_index= static_cast<uint32_t>(index);
```

Comment:
Why are we changing the type here? `index` seems like it can be negative.

Reply:
If `index` is negative, then we would be accessing out-of-bounds into `sine_table` below. Should I `assert(angle >= 0)`?

Even if I undo these casts, I'll still need to do some casting anyways because these indices are being compared to `unsigned int sine_table_size`.